File: payroll-api/src/main/java/com/company/payroll/mapper/StaffLoanMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffLoan;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffLoanMapper {
    int deleteByPrimaryKey(Integer loanId);
    int insert(StaffLoan row);
    int insertSelective(StaffLoan row);
    List<StaffLoan> selectList();
    List<StaffLoan> selectListByStaffId(Integer staffId);
    StaffLoan selectByPrimaryKey(Integer loanId);
    int updateByPrimaryKeySelective(StaffLoan row);
    int updateByPrimaryKey(StaffLoan row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffLoanMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffLoan
import org.springframework.stereotype.Repository

@Repository
interface StaffLoanMapper {
  fun deleteByPrimaryKey(loanId: Int): Int

  fun insert(row: StaffLoan): Int

  fun insertSelective(row: StaffLoan): Int

  fun selectByPrimaryKey(loanId: Int): StaffLoan

  fun selectList(): List<StaffLoan>

  fun selectListByStaffId(staffId: Int): List<StaffLoan>

  fun updateByPrimaryKeySelective(row: StaffLoan): Int

  fun updateByPrimaryKey(row: StaffLoan): Int
}